#167164944 View an Edit Author Profile

OvieMudi opened the pull request:
#### What does this PR do?
- Adds a page for viewing author's profile

#### Description of Task to be completed?
- Ensure route below display the Profile Page
`/profile`
e.g http://localhost:3000/profile

#### How should this be manually tested?
To test manually on your local computer with `localhost`

```
# Clone this repository
$ git clone https://github.com/andela/dahlia-ah-frontend.git

# Navigate to the application folder
$ cd dahlia-ah-frontend

# Checkout the branch
$ git checkout feature/167164944/user-can-view-profile
# Install dependencies
$ npm install

# Start the app in development mode
$ npm run dev
```
- provide the URL of the backend API server as an environment variable
- check the `.env.example` file on the environmental variables to provide

Note: Kindly checkouts the screenshots section the layout to expect
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167164944 ](https://www.pivotaltracker.com/story/show/167164944 )

#### Screenshots
<img width="1440" alt="Screenshot 2019-09-12 at 4 55 35 AM" src="https://user-images.githubusercontent.com/42499892/64753181-2ea5e200-d51a-11e9-96d1-d6f571c395e8.png">
<img width="1197" alt="Screenshot 2019-09-12 at 4 56 11 AM" src="https://user-images.githubusercontent.com/42499892/64753184-2ea5e200-d51a-11e9-9cbb-d7646bc6a273.png">
<img width="1197" alt="Screenshot 2019-09-12 at 4 57 48 AM" src="https://user-images.githubusercontent.com/42499892/64753186-2ea5e200-d51a-11e9-929b-aea4554aa2f0.png">
<img width="1197" alt="Screenshot 2019-09-12 at 4 58 14 AM" src="https://user-images.githubusercontent.com/42499892/64753188-2f3e7880-d51a-11e9-8b64-7de372077d73.png">

OvieMudi committed:
feature(author-profile): add view profile page

- design and create components for viewing profile
- add unit testing

[Starts #167164944]

OvieMudi committed:
add edit name and bio functionality

OvieMudi committed:
fix merge conflict

OvieMudi committed:
remove failing tests

OvieMudi committed:
remove failing tests

OvieMudi committed:
fix conflict with package-lock.json

Nkemjiks commented on src/api/index.js:
```diff
@@ -2,8 +2,11 @@ import axios from 'axios';
-const { token } = JSON.parse(window.localStorage.getItem('AuthorsHavenUser')) || { token: null };
-if (token) {
+
+const user = JSON.parse(window.localStorage.getItem('AuthorsHavenUser'));
```
Why this change?

OvieMudi replied:
This is to handle a case where the AuthorsHavenUser is deleted from the local storage when still logged in (i.e not redirected to the landing page). I faced error when that happened.

Nkemjiks commented on src/api/userProfile.js:
```diff
@@ -0,0 +1,14 @@
+
+const userProfleApi = (userId, setProfileState) => {
+  axios.get(`${BACKEND_PATH}/profiles/${userId}`)
+    .then(({ data }) => {
```
Shouldn't there be a catch block?

OvieMudi committed:
Add profile picture upload and error hadler modal

OvieMudi committed:
remove dotenv for server file

OvieMudi committed:
- fix alignment issue in profile update form
- handle state updates properly to avoid window reload
- add and link "START WRITING"
